src/vs/workbench/parts/tasks/electron-browser/terminalTaskSystem.ts

```diff
@@ -693,6 +692,7 @@ export class TerminalTaskSystem implements ITaskSystem {
 				delete this.terminals[terminalKey];
 				delete this.sameTaskTerminals[terminalData.lastTask];
 				this.idleTaskTerminals.delete(terminalData.lastTask);
+				delete this.activeTasks[Task.getMapKey(task)];
```

Comment:
Would mark this as a work around for the issue since the task should have already been removed if the onExit fires correctly.